ocrs/src/lib.rs

```diff
@@ -128,17 +129,18 @@ impl OcrEngine {
         input: &OcrInput,
         lines: &[Vec<RotatedRect>],
     ) -> Result<Vec<Option<TextLine>>, Box<dyn Error>> {
-        let Some(recognizer) = self.recognizer.as_ref() else {
-            return Err("Recognition model not loaded".into());
-        };
-        recognizer.recognize_text_lines(
-            input.image.view(),
-            lines,
-            RecognitionOpt {
-                debug: self.debug,
-                decode_method: self.decode_method,
-            },
-        )
+        if let Some(recognizer) = self.recognizer.as_ref() {
```

Comment:
This is OK, although if new lines were added to the happy path, I'd probably want to revert back to the early-exit.

Reply:
if I understand correctly, the two versions of the code operate identically and the
`if let Some` vs `let - else` construction merely moves the main body vs exception placements around. This arguably makes it clearer what this part of the code intends to do vs what it does when it fails.

Usually let else is used when there are a lot of following code, where the `if let Some` pattern will introduce braces and tab drift.
Of course, if we assume both operate the same it becomes a matter of choice, and the opinion of the main maintainer/developer matters more IMO

If i inadvertently introduced any change in behavior I think this should also revert back to the original.

Reply:
> Usually let else is used when there are a lot of following code, where the if let Some pattern will introduce braces and tab drift.

Yes, this is one reason. The other is to logically have early-exit checks at the top of the function and then have the main body concentrate on the happy path. This also matters mainly when there are multiple early-exits. With the current body both are easy to read, and I would be fine with either.